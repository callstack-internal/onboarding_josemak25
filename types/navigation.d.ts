import type { NativeStackScreenProps } from "@react-navigation/native-stack";

export type RootStackParamList = {
  Home: undefined;
  Detail: { data: Weather };
};

export type RootTabScreenProps<Screen extends keyof RootStackParamList> =
  NativeStackScreenProps<RootStackParamList, Screen>;
