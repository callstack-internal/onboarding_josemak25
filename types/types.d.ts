interface Weather {
  coord: {
    lon: number;
    lat: number;
  };
  sys: {
    country: string;
    timezone: number;
    sunrise: number;
    sunset: number;
  };
  weather: [
    {
      id: number;
      icon: string;
      main: string;
      description: string;
    },
  ];
  main: {
    temp: number;
    feels_like: number;
    temp_min: number;
    temp_max: number;
    pressure: number;
    humidity: number;
  };
  visibility: number;
  wind: {
    speed: number;
    deg: number;
  };
  clouds: {
    all: number;
  };
  dt: number;
  id: number;
  name: string;
}

interface BasicError {
  error: any;
  data?: any;
  message: string;
}
