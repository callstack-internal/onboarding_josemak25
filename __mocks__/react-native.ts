jest.mock("react-native", () => {
  const RN = jest.requireActual("react-native"); // use original implementation, which comes with mocks out of the box

  // mock modules/components created by assigning to NativeModules
  RN.NativeModules.SourceCode = {
    scriptURL: "http://localhost:3000/",
  };

  return RN;
});
