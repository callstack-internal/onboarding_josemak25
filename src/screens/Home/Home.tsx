import React from "react";
import { SafeAreaView, StatusBar, Text, View } from "react-native";

import { makeUseStyles } from "../../helpers/makeUseStyles";

export const HomeScreen = () => {
  const { styles } = useStyles();

  return (
    <SafeAreaView style={styles.container}>
      <StatusBar />
      <View>
        <Text>Welcome to the home screen:</Text>
      </View>
    </SafeAreaView>
  );
};

const useStyles = makeUseStyles(() => ({
  container: {
    flex: 1,
  },
}));
