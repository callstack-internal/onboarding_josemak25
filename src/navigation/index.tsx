import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";

import { HomeScreen } from "../screens/Home";
import { DetailScreen } from "../screens/Detail";
import { RootStackParamList } from "../../types/navigation";

const Stack = createNativeStackNavigator<RootStackParamList>();

export const Navigation = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="Home" component={HomeScreen} />
        <Stack.Screen name="Detail" component={DetailScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
};
