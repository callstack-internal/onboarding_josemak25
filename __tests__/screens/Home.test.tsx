import "react-native";
import React from "react";

// Note: test renderer must be required after react-native.
import { render } from "@testing-library/react-native";

import { HomeScreen } from "../../src/screens/Home";
import { RootTabScreenProps } from "../../types/navigation";

describe("screens / Home.tsx", () => {
  it("should render correctly", () => {
    const navigate: RootTabScreenProps<"Home">["navigation"]["navigate"] =
      jest.fn();
    const options = { navigation: { navigate } } as RootTabScreenProps<"Home">;

    render(<HomeScreen {...options} />);
  });
});
