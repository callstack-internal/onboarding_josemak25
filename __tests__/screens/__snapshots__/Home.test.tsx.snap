// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components / Weather.tsx should render correctly 1`] = `
<View
  accessibilityLabel="weather card"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 70,
      "opacity": 1,
      "paddingHorizontal": 12,
    }
  }
  testID="weather_card_press"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Image
      accessibilityIgnoresInvertColors={true}
      resizeMode="contain"
      source={
        Object {
          "uri": "https://openweathermap.org/img/wn/04d.png",
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
    <View
      style={
        Object {
          "marginHorizontal": 12,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "600",
            },
            Object {
              "color": "#000",
              "opacity": 0.5,
            },
          ]
        }
      >
        Madrid
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "600",
            },
            Object {
              "color": "#000",
              "opacity": 0.5,
            },
          ]
        }
      >
        broken clouds
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        Object {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#b4d7e3",
          "borderRadius": 50,
          "paddingHorizontal": 18,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        17.9
         °F
      </Text>
    </View>
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 14,
          },
          Object {
            "marginLeft": 12,
            "opacity": 0.5,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`screens / Detail.tsx should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingBottom": 0,
    }
  }
>
  <View
    accessibilityLabel="weather info"
    accessibilityRole="none"
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 70,
        "opacity": 1,
        "paddingHorizontal": 12,
      }
    }
    testID="weather_card_press"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Image
        accessibilityIgnoresInvertColors={true}
        resizeMode="contain"
        source={
          Object {
            "uri": "https://openweathermap.org/img/wn/04d.png",
          }
        }
        style={
          Object {
            "height": 50,
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "marginHorizontal": 12,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "600",
              },
              Object {
                "color": "#000",
                "opacity": 0.5,
              },
            ]
          }
        >
          Madrid
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "600",
              },
              Object {
                "color": "#000",
                "opacity": 0.5,
              },
            ]
          }
        >
          broken clouds
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#b4d7e3",
            "borderRadius": 50,
            "paddingHorizontal": 18,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          17.9
           °F
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#b2b2b2",
        "borderWidth": 0.5,
        "opacity": 0.1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 12,
        "paddingVertical": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 16,
        }
      }
    >
      Humidity
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 16,
        }
      }
    >
      65
      %
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#b2b2b2",
        "borderWidth": 0.5,
        "opacity": 0.1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 12,
        "paddingVertical": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 16,
        }
      }
    >
      Pressure
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 16,
        }
      }
    >
      1016
       hPa
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#b2b2b2",
        "borderWidth": 0.5,
        "opacity": 0.1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 12,
        "paddingVertical": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 16,
        }
      }
    >
      Wind speed
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 16,
        }
      }
    >
      13.89
       mph
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#b2b2b2",
        "borderWidth": 0.5,
        "opacity": 0.1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 12,
        "paddingVertical": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 16,
        }
      }
    >
      Cloud cover
    </Text>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 16,
        }
      }
    >
      75
      %
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#b2b2b2",
        "borderWidth": 0.5,
        "opacity": 0.1,
      }
    }
  />
</View>
`;

exports[`screens / Home.tsx should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingBottom": 0,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListEmptyComponent={[Function]}
    contentContainerStyle={
      Object {
        "flex": 1,
        "paddingBottom": 0,
      }
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginHorizontal": 24,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#000",
              "borderRadius": 43,
              "borderWidth": 2,
              "height": 86,
              "justifyContent": "center",
              "width": 86,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 38,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 20,
              "marginVertical": 12,
              "textAlign": "center",
            }
          }
        >
          No Weather Data
        </Text>
        <Text
          style={
            Object {
              "color": "#000",
              "textAlign": "center",
            }
          }
        >
          Pull-to-refresh
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
